MIGRATION_GUIDE.md

```diff
@@ -1,5 +1,104 @@
 # Migration Guide
 
+## Upgrading from v0.48.0 → v0.49.0
+
+There are deprecations in this update. Please ensure you read this guide thoroughly and prepare your potential
+automated workflows before upgrading.
+
+### Deprecations
+
+- [Organization Member Roles](#organization-member-roles)
+
+#### Organization Member Roles
+
+The `roles` field on the `auth0_organization_member` resource will continue to be available for managing an organization
+member's roles. However, to ensure a smooth transition when we eventually remove the capability to manage roles through
+this field, we recommend proactively migrating to the newly introduced `auth0_organization_member_roles` resource.
+This will help you stay prepared for future changes.
+
+<table>
+<tr>
+<th>Before (v0.48.0)</th>
+<th>After (v0.49.0)</th>
+</tr>
+<tr>
+<td>
+
+```terraform
+resource "auth0_role" "reader" {
+  name = "Reader"
+}
+
+resource "auth0_role" "admin" {
```

Comment:
If we're using "reader"/"writer" example roles in the "after" snippet below, should we use the same to make the examples equivalent and easier to follow?

Reply:
Updated 👍🏻